liquibase-standard/src/main/java/liquibase/changelog/visitor/ValidatingVisitor.java

```diff
@@ -184,7 +184,7 @@ protected void validateChange(ChangeSet changeSet, Database database, Change cha
 
     private boolean areChangeSetAttributesValid(ChangeSet changeSet) {
         boolean authorEmpty = StringUtils.isEmpty(changeSet.getAuthor());
-        boolean idEmpty = StringUtils.isEmpty(changeSet.getId());
+        boolean idEmpty = StringUtils.isBlank(changeSet.getId());
```

Comment:
@abrackx but this is validation all changesets, including existing ones right? IT's not in the ticket description, but ther eis a comment stating: 
`We will not fix changesets that have been already deployed with empty ChangeSetIDs`
So if the changeset was already applied we should not validate it, otherwise the user will have to fix it and will have a changeset mismatch. 
can we do it here?

Reply:
Good catch! I've changed this validation method to have a `strictIdCheck` flag. 

If the changeset has NOT ran or is about to run (like in the case of `runAlways` or `runOnChange`) we will check this "strictly" aka make sure it's not a space. 
If the changeset has already ran then and is not about to run we will use the previous validation and allow spaces.

I think this makes sense to me, let me know what you guys think @StevenMassaro @filipelautert

Reply:
@abrackx  This `shouldValidate` flag is interesting, good find! I almost think  that instead of passing it to the method, we could use it in the if:

```
if (shouldValidate && !areChangeSetAttributesValid(changeSet)) {
```

What do you think?

shouldValidate exists since 2013 and the commit message is interesting:

> Don't validate changesets that have already been run.
> Do validate the checksums of all changesets.
> Do validate changesets that should run even if they have run before.

Method `areChangeSetAttributesValid `was introduced ~2 years ago and seems it ignored this premisse from `shouldValidate `

Reply:
Updated! Removed the `strictIdCheck` from the `areChangeSetAttributesValid` and added the `shouldValidate` flag to the conditional. Same behavior but now we only validate when the change is actually going to be ran.